internal/scheduler/nodeiteration.go

```diff
@@ -172,44 +267,78 @@ func (it *NodeTypeResourceIterator) NextNodeItem() *schedulerobjects.Node {
 	if obj == nil {
 		return nil
 	}
-	nodeItem, ok := obj.(*schedulerobjects.Node)
+	node, ok := obj.(*schedulerobjects.Node)
 	if !ok {
 		panic(fmt.Sprintf("expected *NodeItem, but got %T", obj))
 	}
-	if nodeItem.NodeTypeId != it.nodeType.Id {
-		// The index is sorted by NodeType first.
-		// So we've seen all nodes of this NodeType if this comparison fails.
+	// If it.dominantQueue != *, the index is sorted by node.DominantQueue() first.
+	// Otherwise, the index is sorted by NodeTypeId first.
+	if it.dominantQueue != "*" {
```

Comment:
I'd move * into some constant so that it's obvious what it is in all contexts

Reply:
Fixed